analyses/molecular-subtyping-CRANIO/00-craniopharyngiomas-molecular-subtype.Rmd

```diff
@@ -0,0 +1,180 @@
+# Inputs
+## Read in metadata
+metadata <-
+  read.table(file.path(root_dir, "data", "pbta-histologies.tsv"),
+             header=T,
+             sep='\t') %>%
+  filter(pathology_diagnosis == "Craniopharyngioma")
+## Read in snv consensus mutation data
+snv_df <-
+  read.table(file.path(root_dir, "data", "pbta-snv-consensus-mutation.maf.tsv.gz"),
+             header=T,
+             sep='\t')
+```
+
+# Process Data
+
+## Collecting Tumor Sample Barcodes with Variants of Interest
+
+Collect Tumor_Sample_Barcodes with BRAF V600E mutations.
+
+```{r braf-tsb}
+braf_v600e_tsb <-
+  snv_df %>%
+  filter(Hugo_Symbol == "BRAF" & HGVSp_Short == "p.V600E") %>%
+  select(Tumor_Sample_Barcode)
+```
+
+Collect Tumor_Sample_Barcodes with CTNNB1 mutation on exon 3.
+
+```{r ctnnb1-tsb}
+ctnnb1_e3_tsb <-
+  snv_df %>%
+  filter(Hugo_Symbol == "CTNNB1" & Exon_Number == "3/15") %>%
+  select(Tumor_Sample_Barcode)
+```
+
+## Defining Lesion Table
+
+Annotate the metadata table where the Kids_First_Biospecimen_ID matches either of the TSBs collected above
+Lesions are defined where:
+1. patient has BRAF.V600E mutation and is older than the age_of_adulthood (40 years in this case)
+1. patient has CTNNB1.Exon3 mutation and is younger than the age_of_adulthood (40 years in this case)
+Select the columns from the table that the PIs have requested
+
+```{r snv-lesion-table}
+# Dispute this all you like
+age_of_adulthood <- 40 * 365.25
+
+snv_lesion_df <-
+  metadata %>%
+  mutate(CTNNB1.Exon3 = Kids_First_Biospecimen_ID %in% ctnnb1_e3_tsb$Tumor_Sample_Barcode) %>%
+  mutate(BRAF.V600E = Kids_First_Biospecimen_ID %in% braf_v600e_tsb$Tumor_Sample_Barcode) %>%
+  mutate(defining_lesion =
+           (BRAF.V600E & as.numeric(as.character(age_at_diagnosis_days)) >= age_of_adulthood) |
+           (CTNNB1.Exon3 & as.numeric(as.character(age_at_diagnosis_days)) < age_of_adulthood)) %>%
+  select(Kids_First_Participant_ID,
+         sample_id,
+         Kids_First_Biospecimen_ID,
+         age_at_diagnosis_days,
+         CTNNB1.Exon3,
+         BRAF.V600E,
+         defining_lesion)
+```
+
+## Defining Molecular Subtype Table
+
+Make a frame with DNA and RNA samples per patient
+
+```{r allmeta-table}
+dnameta <-
+  metadata %>%
+  filter(experimental_strategy == c("WGS","WXS")) %>%
+  select(Kids_First_Participant_ID,sample_id,Kids_First_Biospecimen_ID)
+
+rnameta <-
+  metadata %>%
+  filter(experimental_strategy == "RNA-Seq") %>%
+  select(Kids_First_Participant_ID,sample_id,Kids_First_Biospecimen_ID)
+
+allmeta <-
+  full_join(dnameta,
+            rnameta,
+            by=c("Kids_First_Participant_ID","sample_id"),
+            suffix = c("_DNA", "_RNA"))
+```
+
+Append stringified lesion information to the table
+Select the columns from the table that the PIs have requested
+
+```{r meta-subtype-table}
+snv_subtype <-
+  snv_lesion_df %>%
+  mutate(molecular_subtype =
+           case_when(
+             defining_lesion & BRAF.V600E & !CTNNB1.Exon3 ~ "CRANIO, PAP",
```

Comment:
Because logic things can be brain teasers, I just like to make sure I'm following what's happening. 

`"CRANIO, PAP"` are samples that are older than 40yrs and ONLY have BRAF.V600E mutations 
`"CRANIO, ADAM"` are samples that are older than 40yrs and ONLY have CTNNB1.Exon3 mutations
`"CRANIO, ADAM;CRANIO, PAP"` are samples with BOTH mutations and are older than 40
`"CRANIO, To be classified"` means they aren't older than 40 OR they don't have either mutation 

I think `"CRANIO, To be classified"` is least clear to me since there are multiple reasons that this might be unclassified.

Reply:
I've added explicit definitions to the text block preceding this command:
```
Append molecular subtypes defined as follows:
1. `CRANIO, PAP`: patient has BRAF.V600E mutation and is older than the age_of_adulthood (40 years in this case)
1. `CRANIO, ADAM`:patient has CTNNB1.Exon3 mutation and is younger than the age_of_adulthood (40 years in this case)
1. `CRANIO, ADAM;CRANIO, PAP` Both of the above (this is actually impossible given the restriction on age)
1. `CRANIO, To be classified` None of the above
```

I say it in the block but `CRANIO, ADAM;CRANIO, PAP` is impossible given my current restrictions using age. Would it make sense to remove the age restriction from the `ADAM` check since as the ticket says "but can be seen in adults"?

Reply:
Thanks for this summary, this is great. Can you explain what you mean by  is actually impossible given the restriction on age`? 

And I guess my follow up question is, do we want this to be an impossible category to have?

Reply:
Well you can only be CRANIO, ADAM if your age at diagnosis is under the age of adulthood and you can only be CRANIO, PAP if your age at diagnosis is at or above the age of adulthood. Since age_at_diagnosis can can only be one value, its an either or.

Reply:
Okay. Gotcha. Thanks for this explanation, I don't know about how this calls are made. In this instance then it seems this logic fits what you are describing. :+1:

Reply:
I asked the person who assigned this to me to clarify the restriction. I'll update if they can confirm one way or the other.